### PR TITLE
hugo/feature/Refactor objects into namespaces

### PR DESCRIPTION
**Pull request by HPezz**
- :art: (reinforcer): Refactor objects into namespaces
- :art: (ledkit): Refactor objects into namespaces
- :art: (colorkit): Refactor objects into namespaces
- :art: (behaviorkit): Refactor objects into namespaces
- :art: (os): Refactor objects into namespaces
- :art: (bootloader): Refactor objects into namespaces
- :art: (bootloader): Refactor globals into namespace/anonymous namespace

### CODE REVIEW

**Review by ladislas — APPROVED** [review]

File: spikes/lk_led_animations/main.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			auto event_queue = events::EventQueue {};
+
+		}	// namespace animations
+
+	}	// namespace internal
+
+	auto ears = CoreLED<internal::ears::size> {internal::ears::spi};
+	auto belt = CoreLED<internal::belt::size> {internal::belt::spi};
+
+	auto kit = LedKit {internal::animations::thread, internal::animations::event_queue, ears, belt};
```
Comment (YannLocatelli) [inline spikes/lk_led_animations/main.cpp:60]: This refactorization is different from `spike/lk_led_kit`, definition of namespace and their orders risk to be inconstant among different `main.cpp`.

```cpp
namespace leds {

namespace spi {

	auto belt = CoreSPI {LED_BELT_SPI_MOSI, NC, LED_BELT_SPI_SCK};
	auto ears = CoreSPI {LED_EARS_SPI_MOSI, NC, LED_EARS_SPI_SCK};

}	// namespace spi

namespace animations {

	auto thread		 = rtos::Thread {};
	auto event_queue = events::EventQueue {};

}	// namespace animations

auto ears = CoreLED<LedKit::kNumberOfLedsEars> {spi::ears};
auto belt = CoreLED<LedKit::kNumberOfLedsBelt> {spi::belt};

}	// namespace leds
```

File: spikes/lk_led_kit/main.cpp [hunk 2]
```diff
@@ -41,16 +41,28 @@
 using namespace leka;
 using namespace std::chrono;
 
-auto corespi_belt = CoreSPI {LED_BELT_SPI_MOSI, NC, LED_BELT_SPI_SCK};
-auto corespi_ears = CoreSPI {LED_EARS_SPI_MOSI, NC, LED_EARS_SPI_SCK};
+namespace leds {
```
Comment (YannLocatelli) [inline spikes/lk_led_kit/main.cpp:44]: Missing unnamed namespace

File: spikes/lk_behavior_kit/main.cpp [hunk 3]
```diff
@@ -22,51 +22,128 @@
...
+	namespace internal {
 
-auto motor_left_dir_1  = mbed::DigitalOut {MOTOR_LEFT_DIRECTION_1};
-auto motor_left_dir_2  = mbed::DigitalOut {MOTOR_LEFT_DIRECTION_2};
-auto motor_left_speed  = CorePwm {MOTOR_LEFT_PWM};
-auto motor_right_dir_1 = mbed::DigitalOut {MOTOR_RIGHT_DIRECTION_1};
-auto motor_right_dir_2 = mbed::DigitalOut {MOTOR_RIGHT_DIRECTION_2};
-auto motor_right_speed = CorePwm {MOTOR_RIGHT_PWM};
+		auto bd = SDBlockDevice {SD_SPI_MOSI, SD_SPI_MISO, SD_SPI_SCK};
+		auto fs = FATFileSystem {"fs"};
```
Comment (YannLocatelli) [inline spikes/lk_behavior_kit/main.cpp:36]: Not a big fan of acronyms, except for the well-known like WiFi or SD, we should use an explicit name or use a glossary

File: app/bootloader/main.cpp [hunk 4]
```diff
@@ -24,115 +24,199 @@
...
+}	// namespace os
 
-auto motor_left	 = CoreMotor {motor_left_dir_1, motor_left_dir_2, motor_left_speed};
-auto motor_right = CoreMotor {motor_right_dir_1, motor_right_dir_2, motor_right_speed};
+namespace battery {
 
-void initializeSD()
-{
-	constexpr auto default_sd_blockdevice_frequency = uint64_t {25'000'000};
+	namespace status {
```
Comment (YannLocatelli) [inline app/bootloader/main.cpp:47]:
```suggestion
	namespace charge_status {
```
or 
```suggestion
	namespace charge {
```

File: app/os/main.cpp [hunk 5]
```diff
@@ -30,133 +30,255 @@
...
+//
 
-auto &instance		   = mbed::Watchdog::get_instance();
-constexpr auto timeout = 30000ms;
-auto thread			   = rtos::Thread {osPriorityLow};
+namespace {
+
+namespace battery {
+
+	namespace status {
```
Comment (YannLocatelli) [inline app/os/main.cpp:41]:
```suggestion
	namespace charge {
```

File: app/os/main.cpp [hunk 6]
```diff
@@ -30,133 +30,255 @@
...
 
-}	// namespace watchdog
+}	// namespace battery
+
+namespace sd {
+
+	namespace internal {
+
+		auto bd = SDBlockDevice {SD_SPI_MOSI, SD_SPI_MISO, SD_SPI_SCK};
+		auto fs = FATFileSystem {"fs"};
```
Comment (YannLocatelli) [inline app/os/main.cpp:56]: acronyms

File: app/os/main.cpp [hunk 7]
```diff
@@ -30,133 +30,255 @@
...
+
+auto videokit	 = VideoKit {};
+auto behaviorkit = BehaviorKit {videokit, leds::kit, motors::left::motor, motors::right::motor};
 
 namespace command {
 
-namespace internal {
+	namespace internal {
+
+		auto test		= TestCommand {};
```
Comment (YannLocatelli) [inline app/os/main.cpp:158]: Out of this PR, is this used?

**Review by YannLocatelli — COMMENTED** [review]
There is quite much modifications to review

**Review by YannLocatelli — APPROVED** [review]
That's good to me

**Comment by ladislas** [comment]
closed by fe54bcf9fb48676617f09a24f687e6e8d3b7f964

**Pull request closed without merging**